libbeat/processors/add_process_metadata/gosigar_cid_provider.go

```diff
@@ -106,25 +107,35 @@ func (p gosigarCidProvider) getProcessCgroups(pid int) (map[string]string, error
 // ID is found then an empty string is returned.
 // Example:
 // /kubepods/besteffort/pod9b9e44c2-00fd-11ea-95e9-080027421ddf/2bb9fd4de339e5d4f094e78bb87636004acfe53f5668104addc761fe4a93588e
-func (p gosigarCidProvider) getCid(cgroups map[string]string) string {
+func (p gosigarCidProvider) getCid(cgroups map[string]cgroup.ControllerPath) string {
 	// if regex defined use it to find cid
 	if len(p.cgroupRegex) != 0 {
 		re := regexp.MustCompile(p.cgroupRegex)
 		for _, path := range cgroups {
-			rs := re.FindStringSubmatch(path)
+			rs := re.FindStringSubmatch(path.ControllerPath)
 			if rs != nil {
 				return rs[1]
 			}
 		}
 	} else {
-		// use string prefix to find cid
+		// else, fall back to a hardcoded regex
+		// In an attempt to not break the user-facing config interface for this processor,
+		// fall back to the config'ed prefixes if we have cgv1, otherwise use regex
```

Comment:
I see `p.cgroupPrefixes` are not used if this is using cgroups v2. This may be a bit unexpected, or this option doesn't make sense under cgroup v2?

Reply:
The prefixes are used for different controllers within the cgroup. On V2, the controllers are "unified" so they aren't listed in `/proc/PID/cgroup` We might be able to hack around to make the option do something, but it doesn't really make sense here, I think.